lib/evse_security/evse_security.cpp

```diff
@@ -513,11 +513,21 @@ void EvseSecurity::update_ocsp_cache(const CertificateHashData& certificate_hash
 
 bool EvseSecurity::is_ca_certificate_installed(CaCertificateType certificate_type) {
     try {
-        X509Wrapper(this->ca_bundle_path_map.at(certificate_type), EncodingFormat::PEM);
-        return true;
+        X509CertificateBundle bundle(this->ca_bundle_path_map.at(certificate_type), EncodingFormat::PEM);
+
+        // Search for a valid self-signed root
+        auto& hierarchy = bundle.get_certficate_hierarchy();
+
+        // Get all roots and search for a valid self-signed
+        for (auto& root : hierarchy.get_hierarchy()) {
+            if (root.certificate.is_selfsigned() && root.certificate.is_valid())
+                return true;
+        }
     } catch (const CertificateLoadException& e) {
         return false;
```

Comment:
Glad to see this is being fixed, about to make a PR for this as well.

Maybe you could add an error print in the catch so we don't silently fail.